### PR TITLE
Add golang to pipelines

### PR DESCRIPTION
**Pull request by william-msft**
- [ ] The purpose of this PR is explained in this message or in an issue. If an issue please include a reference as https://devdiv.visualstudio.com/DevDiv/_boards/board/t/Oryx/Stories/?workitem=1531667
- [ ] Tests are included and/or updated for code changes.
- [ ] Proper license headers are included in each file.

### CODE REVIEW

**Review by daniv-msft — DISMISSED** [review]
Looks good!

File: tests/Oryx.BuildImage.Tests/golang/GolangDynamicInstallationTest.cs [hunk 1]
```diff
@@ -13,7 +13,7 @@
 
 namespace Microsoft.Oryx.BuildImage.Tests
 {
-    [Trait("platform", "golang")]
+    [Trait("platform", "golang-dynamic")]
```
Comment (qianz2) [inline tests/Oryx.BuildImage.Tests/golang/GolangDynamicInstallationTest.cs:16]: Is this golang dynamic installation test used for go `full` image, should we name it as `golang-full` or other names?

File: vsts/pipelines/templates/_integrationJobTemplate.yml [hunk 2]
```diff
@@ -31,6 +31,10 @@ parameters:
       - 60
       - dynamic
       - runtime
+  - name: golangVersions
+    type: object
+    default:
+      - dynamic
```
Comment (qianz2) [inline vsts/pipelines/templates/_integrationJobTemplate.yml:37]: same here.

File: vsts/scripts/tagBuildImagesForRelease.sh [hunk 3]
```diff
@@ -54,6 +54,7 @@ tagBuildImage "oryxdevmcr.azurecr.io/public/oryx/build:github-actions-$BUILD_DEF
 tagBuildImage "oryxdevmcr.azurecr.io/public/oryx/build:github-actions-buster-$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME" "github-actions-buster" "github-actions-buster-$RELEASE_TAG_NAME"
 tagBuildImage "oryxdevmcr.azurecr.io/public/oryx/build:github-actions-bullseye-$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME" "github-actions-bullseye" "github-actions-bullseye-$RELEASE_TAG_NAME"
 tagBuildImage "oryxdevmcr.azurecr.io/public/oryx/build:vso-focal-$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME" "vso-focal" "vso-focal-$RELEASE_TAG_NAME"
+tagBuildImage "oryxdevmcr.azurecr.io/public/oryx/build:full-$BUILD_DEFINITIONNAME.$RELEASE_TAG_NAME" "full" "full-$RELEASE_TAG_NAME"
```
[the remark this answers is not in the document]
Comment (william-msft) [reply vsts/scripts/tagBuildImagesForRelease.sh]: `full-buster` sounds good to me. I've made the change, thanks!

File: tests/Oryx.BuildImage.Tests/golang/GolangDynamicInstallationTest.cs (hunk 1 above)
Comment (william-msft) [reply tests/Oryx.BuildImage.Tests/golang/GolangDynamicInstallationTest.cs]: Sounds good to me, I've made the change. Thanks!

File: vsts/pipelines/templates/_integrationJobTemplate.yml (hunk 2 above)
Comment (william-msft) [reply vsts/pipelines/templates/_integrationJobTemplate.yml]: Sounds good to me, I've made the change. Thanks!

**Review by qianz2 — APPROVED** [review]

**Pull request merged**